### PR TITLE
Add the mask workflow

### PR DESCRIPTION
**Pull request by matthieudumont**

### CODE REVIEW

File: dipy/workflows/mask.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        out_mask : string, optional
+           Name of the masked file (default 'mask.nii.gz')
+        """
+        io_it = self.get_io_iterator()
+
+        for input_path, out_mask_path in io_it:
+            logging.info('Creating mask of {0}'
+                         .format(input_path))
+            data, affine = load_nifti(input_path)
+            mask = np.bitwise_and(data > greater_than, data < less_than)
```
Comment (arokem) [inline dipy/workflows/mask.py]: Maybe error handling for the case where greater_than > less_than?

And testing for that case

File: dipy/workflows/mask.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            out_mask='mask.nii.gz'):
+
+        """ Workflow for creating a binary mask
+
+        Parameters
+        ----------
+        input_files : string
+           Path to image to be masked.
+        greater_than : float
+            Default is 0.2.
```
Comment (arokem) [inline dipy/workflows/mask.py]: What's the 0.2 default all about?

**Comment by Garyfallidis** [comment]
@matthieudumont what is the purpose of this PR?

**Comment by matthieudumont** [comment]
@Garyfallidis : Add the mask workflow

**Comment by Garyfallidis** [comment]
@matthieudumont I noticed something while playing with the current master. In setup.py the script dipy_reconst_dti_restore is not available. The others are. You may want to add it here together with dipy_mask.

**Comment by matthieudumont** [comment]
Good catch, will add.

**Comment by arokem** [comment]
And add a test that these things are installed and can run, please.

File: dipy/workflows/mask.py (hunk 2 above)
Comment (matthieudumont) [inline dipy/workflows/mask.py]: I guess this should not be an optional param after all, no idea where 0.2 came from when I originally did this.

File: dipy/workflows/mask.py (hunk 1 above)
Comment (matthieudumont) [inline dipy/workflows/mask.py]: Good point, will do.

**Comment by matthieudumont** [comment]
About adding tests that all the available workflows are present and runable, I will make this another PR.

File: dipy/workflows/mask.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ class MaskFlow(Workflow):
     def get_short_name(cls):
         return 'mask'
 
-    def run(self, input_files, greater_than=0.2, less_than=np.inf, out_dir='',
+    def run(self, input_files, greater_than, less_than=np.inf, out_dir='',
```
Comment (arokem) [inline dipy/workflows/mask.py]: How about `ub` and `lb` (for 'upper bound' and 'lower bound')?

File: setup.py [hunk 4]
```diff
@@ -214,6 +214,8 @@ def main(**extra_args):
           data_files=[('share/doc/dipy/examples',
                        glob(pjoin('doc', 'examples','*.py')))],
           scripts      = [pjoin('bin', 'dipy_reconst_dti'),
+                          pjoin('bin', 'dipy_reconst_dti_restore'),
+                          pjoin('bin', 'dipy_mask'),
```
Comment (arokem) [inline setup.py]: This all seems pretty error prone. Is there some way to automate this, instead of having to add a line in here every time?

Comment (matthieudumont) [inline setup.py]: I can look into that in the upcoming script tests PR.

**Comment by Garyfallidis** [comment]
This looks good. Will merge in 2 days.

**Comment by matthieudumont** [comment]
bump! :)

**Comment by Garyfallidis** [comment]
Sorry for the delay!

**Pull request merged**